lib/internal/modules/cjs/loader.js

```diff
@@ -1858,3 +2040,4 @@ ObjectDefineProperty(Module.prototype, 'constructor', {
 
 // Backwards compatibility
 Module.Module = Module;
+Module.registerHooks = registerHooks;
```

Comment:
Because `require('module') === Module`, `module.register` is also added to `Module`

https://github.com/nodejs/node/blob/e312d60e3f81dfa26d5922f3c23a371b2b2ce343/lib/module.js#L19

it's just a matter of adding it in `modules/cjs/loader.js` directly, or importing it and adding it and re-exporting it in `module.js`.

https://github.com/nodejs/node/blob/e312d60e3f81dfa26d5922f3c23a371b2b2ce343/lib/module.js#L28